### PR TITLE
Replacing KillProcess with taskkill /t which kills entire process tree

### PR DESCRIPTION
**Pull request by moozzyk**

### CODE REVIEW

File: src/Microsoft.AspNetCore.Server.Testing/Deployers/ApplicationDeployer.cs [hunk 1]
```diff
@@ -123,6 +122,35 @@ protected void ShutDownIfAnyHostProcess(Process hostProcess)
...
+        private void KillProcess(int? processId)
+        {
+            if (processId == null)
+            {
+                return;
+            }
+
+            ProcessStartInfo startInfo;
+
+            if (PlatformServices.Default.Runtime.OperatingSystem.Equals("Windows", StringComparison.OrdinalIgnoreCase))
```
Comment (victorhurdugaci) [inline src/Microsoft.AspNetCore.Server.Testing/Deployers/ApplicationDeployer.cs]: This will never change during a test run :) Cache it

Comment (moozzyk) [inline src/Microsoft.AspNetCore.Server.Testing/Deployers/ApplicationDeployer.cs]: OK

File: src/Microsoft.AspNetCore.Server.Testing/Deployers/ApplicationDeployer.cs [hunk 2]
```diff
@@ -106,8 +106,7 @@ protected void ShutDownIfAnyHostProcess(Process hostProcess)
             if (hostProcess != null && !hostProcess.HasExited)
             {
                 // Shutdown the host process.
-                hostProcess.Kill();
-                hostProcess.WaitForExit(5 * 1000);
+                KillProcess(hostProcess.Id);
```
Comment (victorhurdugaci) [inline src/Microsoft.AspNetCore.Server.Testing/Deployers/ApplicationDeployer.cs]: This can never be null

**Comment by victorhurdugaci** [comment]
:shipit:

**Comment by moozzyk** [comment]
:up::date:

**Pull request merged**